src/configuration/WithTemplateConfig/WithTemplatesSettings.tsx

```diff
@@ -0,0 +1,106 @@
+import { groupBy } from 'lodash';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import { DataSourcePluginOptionsEditorProps } from "@grafana/data";
+import { HorizontalGroup, Icon, Tab, TabContent, TabsBar, useStyles2 } from "@grafana/ui";
+
+import { PromOptions } from "../../types";
+
+import DashboardItem from "./DasboardItem/DashboardItem";
+import WithTemplateTips from "./WithTemplateTips/WithTemplateTips";
+import getDashboardList from "./api/getDashboardList";
+import { generalFolderTitle, infoTab } from "./constants";
+import getStyles from "./style";
+import { DashboardType, WithTemplate } from "./types";
+
+type Props = DataSourcePluginOptionsEditorProps<PromOptions>;
+const WithTemplatesSettings: FC<Props> = (props) => {
+  const { options, onOptionsChange } = props;
+
+  const styles = useStyles2(getStyles);
+
+  const templates = useMemo(() => options.jsonData.withTemplates || [], [options.jsonData])
+
+  const [dashboards, setDashboards] = useState<Record<string, DashboardType[]>>({})
+  const [openFolder, setOpenFolder] = useState(generalFolderTitle)
+  const [loading, setLoading] = useState(false)
+
+  const handleSaveTemplate = (template: WithTemplate) => {
+    const oldTemplates = templates.filter(t => t.uid !== template.uid)
+    const withTemplates = [...oldTemplates, template]
+    onOptionsChange({
+      ...options,
+      jsonData: {
+        ...options.jsonData,
+        withTemplates,
+      },
+    })
+  }
+  const handleChangeTab = (folderTitle: string) => (e?: React.MouseEvent<HTMLAnchorElement>) => {
+    e?.preventDefault()
+    setOpenFolder(folderTitle)
+  }
+
+  useEffect(() => {
+    const fetchDashboards = async () => {
+      setLoading(true)
+      try {
+        const dashboardsData = await getDashboardList()
+        const groupByFolder = groupBy(dashboardsData, 'folderTitle')
+        setDashboards(groupByFolder)
+      } catch (error) {
+        console.error('Error fetching dashboards:', error);
+      }
+      setLoading(false)
+    }
+
+    fetchDashboards()
+  }, [])
```

Comment:
Since this is within the data source configuration section, there is no information about dashboards in the props. You can move the selection process to the parent component (`ConfigEditor`), but that information is not relevant there as it is only used in `WithTemplateConfig`.